packages/now-node/src/helpers.ts

```diff
@@ -72,6 +73,46 @@ function getCookieParser(req: NowRequest) {
   };
 }
 
+function parseQSArray(key: string, value: any, acc: any): void {
+  const bktNotation = /\[[^[\]\d]?]/.test(key);
+  const result = /\[(\d*)\]$/.exec(key);
+  key = key.replace(/\[\d*\]$/, '');
+
+  if (!result) {
+    acc[key] = value;
+    return;
+  }
+
+  if (acc[key] === undefined) {
+    if (bktNotation === true) {
+      acc[key] = value;
+      return;
+    } else {
+      acc[key] = [value];
+    }
+  }
+
+  if (bktNotation === true) {
+    acc[key] = [].concat(acc[key], value);
+  } else {
+    acc[key][result[1]] = value;
+  }
+}
+
+function parseQS(input: { [key: string]: string }) {
+  const parsed = Object.create(null);
+
+  for (const [param, value] of Object.entries(input)) {
+    parseQSArray(param, value, parsed);
+  }
+
+  for (const [key, value] of Object.entries(parsed)) {
+    parsed[key] = value;
+  }
+
+  return parsed;
+}
+
```

Comment:
Please add tests to `test/helpers.test.js` file. You'll see a `req.query` section in there.